src/main/java/com/microsoft/sqlserver/jdbc/ParameterMetaDataCache.java

```diff
@@ -54,15 +54,14 @@ static boolean getQueryMetadata(Parameter[] params, ArrayList<String> parameterN
         }
 
         for (int i = 0; i < params.length; i++) {
-            boolean found = metadataMap.containsKey(parameterNames.get(i));
             CryptoMetadata foundData = metadataMap.get(parameterNames.get(i));
 
             /*
-             * If ever the map doesn't contain a parameter, the cache entry cannot be used. If there is data found, it
-             * should never have the initialized algorithm as that would contain the key. Clear all metadata that has
-             * already been assigned in either case.
+             * It's fine for the cache to not contain a parameter. This means the parameter uses a plaintext encryption
+             * and its CryptoMetadata was never formed. It is NOT okat for there to be found data with an initialized

```

Comment:
typo also when would this ever happen is this an error condition should we fail or at least log a warning or something?

Reply:
I can add a warning. I don't think its a failure as its working correctly; cache parameter encryption values, no encryption, nothing to cache. But I agree with logging a warning.